lib/p2p/Peer.ts

```diff
@@ -174,25 +189,26 @@ class Peer extends EventEmitter {
    * @param nodePubKey the expected nodePubKey of the node we are opening a connection with
    * @param retryConnecting whether to retry to connect upon failure
    */
-  public open = async (handshakeData: HandshakeState, nodePubKey?: string, retryConnecting = false): Promise<void> => {
+  public open = async (nodeState: NodeState, nodeKey: NodeKey, expectedNodePubKey?: string, retryConnecting = false): Promise<void> => {
```

Comment:
For consistency (and improved clarity) we can call `nodeState` `ownNodeState` as you're done elsewhere.